respect-lib-shared/src/commonMain/kotlin/world/respect/shared/viewmodel/clazz/student/StudentViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package world.respect.shared.viewmodel.clazz.student
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import org.jetbrains.compose.resources.getString
+import world.respect.shared.generated.resources.Res
+import world.respect.shared.viewmodel.RespectViewModel
+import world.respect.shared.generated.resources.add_student
+
+
+data class AddStudentUIState(
+    val name:String=""
+)
+class StudentViewModel(
```

Comment:
Should be AddPersonToClazz - the name of the viewmodel, screen, and ui state class must follow the pattern.